tracer/test/Datadog.Trace.Tests/Util/Http/QueryStringObfuscatorTests.cs

```diff
@@ -63,15 +63,26 @@ public void ObfuscateWithCustomPattern()
             result.Should().Be("?<redacted>&token=a0b21ce2-006f-4cc6-95d5-d7b550698482&key2=val2&<redacted>&<redacted>");
         }
 
-        [Fact]
-        public void ObfuscateTimeout()
+        [Theory]
+        [InlineData(false)]
+#if NETCOREAPP3_1_OR_GREATER
+        [InlineData(true)]
+#endif
+        public void ObfuscateTimeout(bool supportsNonBacktracking)
         {
             var logger = new Mock<IDatadogLogger>();
-            var obfuscator = ObfuscatorFactory.GetObfuscator(100, @"\[(.*?)\]((?:.\s*)*?)\[\/\1\]", logger.Object);
+            var obfuscator = ObfuscatorFactory.GetObfuscator(100, @"\[(.*?)\]((?:.\s*)*?)\[\/\1\]", logger.Object, supportsNonBacktracking: supportsNonBacktracking);
             const string queryString = @"?[tag1]Test's Text Test Text Test Text Test Text.Test Text Test Text Test ""Text Test Text"" Test Text Test Text.Test Text ? Test Text Test Text Test Text Test Text Test Text Test Text.Test Text, Test Text Test Text Test Text Test Text Test Text Test Text Test Text.[/ ta.g1][tag2]Test's Text Test Text Test Text Test Text.Test Text Test Text Test ""Text Test Text"" Test Text Test Text.Test Text ? Test Text Test Text Test Text Test Text Test Text Test Text.Test Text, Test Text Test Text Test Text Test Text Test Text Test Text Test Text.[/ tag2]";
-            var result = obfuscator.Obfuscate(queryString);
-            result.Should().Be(string.Empty);
-            logger.Verify(o => o.Error(It.IsAny<RegexMatchTimeoutException>(), It.IsAny<string>(), It.IsAny<double>(), It.IsAny<string>(), It.IsAny<int>(), It.IsAny<string>()), Times.Once);
+            if (supportsNonBacktracking)
+            {
+                Assert.Throws<NotSupportedException>(() => obfuscator.Obfuscate(queryString));
+            }
+            else
+            {
+                var result = obfuscator.Obfuscate(queryString);
+                result.Should().Be(string.Empty);
+                logger.Verify(o => o.Error(It.IsAny<NotSupportedException>(), It.IsAny<string>(), It.IsAny<double>(), It.IsAny<string>(), It.IsAny<int>(), It.IsAny<string>()), Times.Once);
```

Comment:
```suggestion
                logger.Verify(o => o.Error(It.IsAny<RegexMatchTimeoutException>(), It.IsAny<string>(), It.IsAny<double>(), It.IsAny<string>(), It.IsAny<int>(), It.IsAny<string>()), Times.Once);
```

Reply:
Thanks @anna-git ! Can't seem to do anything right today 😢